Sample Applications/DataBindingDemo/AddProductWindow.cs

```diff
@@ -26,11 +26,34 @@ private void OnInit(object sender, RoutedEventArgs e)
                 SpecialFeatures.None);
         }
 
+        private void AnnounceError(string message)
+        {
+            ErrorTextBlock.Visibility = Visibility.Visible;
+            ErrorTextBlock.Text = message;
+            var automationPeer = UIElementAutomationPeer.CreatePeerForElement(ErrorTextBlock);
```

Comment:
Instead call
```
var automationPeer = ErrorTextBlock.GetAutomationPeer();
```
Your call creates a peer even when there is no automation client.

Reply:
Sorry, that's internal.   The real pattern is to check whether anyone's listening to the event before doing the work.  For example, from Button.OnClick:
```
            if (AutomationPeer.ListenerExists(AutomationEvents.InvokePatternOnInvoked))
            {
                AutomationPeer peer = UIElementAutomationPeer.CreatePeerForElement(this);
                if (peer != null)
                    peer.RaiseAutomationEvent(AutomationEvents.InvokePatternOnInvoked);
            }
```